src/core/thread/mle.cpp

```diff
@@ -1480,21 +1473,30 @@ void Mle::HandleParentRequestTimer(void)
             switch (mParentRequestMode)
             {
             case kAttachAny:
-                if (mPreviousPanId != Mac::kPanIdBroadcast)
-                {
-                    netif.GetMac().SetChannel(mPreviousChannel);
-                    netif.GetMac().SetPanId(mPreviousPanId);
-                    mPreviousPanId = Mac::kPanIdBroadcast;
-                    BecomeDetached();
-                }
-                else if ((mDeviceMode & ModeTlv::kModeFFD) == 0)
+                if (mRole != OT_DEVICE_ROLE_CHILD)
                 {
-                    SendOrphanAnnounce();
-                    BecomeDetached();
+                    if (mPreviousPanId != Mac::kPanIdBroadcast)
+                    {
+                        netif.GetMac().SetChannel(mPreviousChannel);
+                        netif.GetMac().SetPanId(mPreviousPanId);
+                        mPreviousPanId = Mac::kPanIdBroadcast;
+                        BecomeDetached();
+                    }
+                    else if ((mDeviceMode & ModeTlv::kModeFFD) == 0)
+                    {
+                        SendOrphanAnnounce();
+                        BecomeDetached();
+                    }
+                    else if (netif.GetMle().BecomeLeader() != OT_ERROR_NONE)
+                    {
+                        BecomeDetached();
+                    }
                 }
-                else if (netif.GetMle().BecomeLeader() != OT_ERROR_NONE)
+                else if ((mDeviceMode & ModeTlv::kModeRxOnWhenIdle) == 0)
                 {
-                    BecomeDetached();
+                    // return to sleepy operation
+                    netif.GetMeshForwarder().GetDataPollManager().SetAttachMode(false);
+                    netif.GetMeshForwarder().SetRxOnWhenIdle(false);
```

Comment:
I think we need to set `mParentRequestState` to `kParentIdle` so that next time `BecomeChild()` is called it does not return `BUSY`. Here is what we changed (expanding the `else if`)...
```                     
                             BecomeDetached();
                    }
                    else
                    {
                        mParentRequestState = kParentIdle;
                        
                        // return to sleepy operation
                        if ((mDeviceMode & ModeTlv::kModeRxOnWhenIdle) == 0)
                        {
                            netif.GetMeshForwarder().GetDataPollManager().SetAttachMode(false);
                            netif.GetMeshForwarder().SetRxOnWhenIdle(false);
                        }
                    }
```

Reply:
Not needed in this PR.  See [L1448](https://github.com/openthread/openthread/pull/2337/files#diff-84a897c448e39fcbd8f2c6c8b49b650bR1448).

Reply:
Indeed... You are right!  Another cherry-pick inconsistency...